src/common/perf_histogram.h

```diff
@@ -0,0 +1,213 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2017 OVH
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#ifndef CEPH_COMMON_PERF_HISTOGRAM_H
+#define CEPH_COMMON_PERF_HISTOGRAM_H
+
+#include "common/Formatter.h"
+#include "include/atomic.h"
+#include "include/int_types.h"
+
+#include <memory>
+
+class PerfHistogramCommon {
+public:
+  enum scale_type_d : uint8_t {
+    SCALE_LINEAR = 1,
+    SCALE_LOG2 = 2,
+  };
+
+  struct axis_config_d {
+    const char *m_name;
+    scale_type_d m_scale_type;
+    int64_t m_min;
+    int64_t m_quant_size;
+    int32_t m_buckets;
```

Comment:
@tchaikov Seems like a great idea.
Those values would also create invalid configuration object and would [assert](https://github.com/ceph/ceph/pull/12829/files#diff-ec4d679fae3103eccf0ed50be360c946R68). This would show up quickly when starting OSD.